src/game/logic/map/polygontrigger.cpp

```diff
@@ -128,8 +128,11 @@ void PolygonTrigger::Update_Bounds() const
     }
 
     m_boundsNeedsUpdate = false;
-    float x = (float)(m_bounds.hi.x - m_bounds.lo.x) / 2.0f;
-    float y = (float)(m_bounds.lo.y + m_bounds.hi.y) / 2.0f;
+
+    // #BUGFIX Calculate correct height value here.
+    float x = (float)(m_bounds.Width()) / 2.0f;

```

Comment:
you can drop the casts. Make the comment `Bugfix Vanilla calculated midpoint for y instead of height`

Reply:
Ok no problem. I reworded the comment better to explain what is happening and refactored the code so it is easier to understand. Removed C casts.